pennylane_forest/qpu.py

```diff
@@ -153,6 +155,15 @@ def expval(self, observable):
         wires = observable.wires
         # Single-qubit observable
         if len(wires) == 1:
+
+            # In case the observable is a Tensor, then
+            # it is made up of exactly one observable
+            # so we can extract it to perform operator
+            # estimation
+            if isinstance(observable, Tensor):
+                observable = observable.obs[0]
+                wires = observable.wires
```

Comment:
Is it intentional that this definition of `wires` gives a list of list, while the one appearing directly above the `if len(wires)==1` condition gives just a list? I'm wondering both (a) whether the list of list structure messes with the intended manipulations below, and (b) if we can just continue to use the already defined `wires`.

Reply:
This is an excellent question! We have decided to address this in PennyLane core and merged the following PR:
https://github.com/XanaduAI/pennylane/pull/498

This PR essentially entails the type of logic contained in the commented lines and treats another edge case:
a) if `Tensor` contains a single observable, then this single observable is returned
b) it also removes any and all `Identity` observables in a `Tensor`. The reason for this is that a `Tensor` could contain *exactly one* non-Identity observable, and yet behave as a `Tensor` containing multiple such observables. This would then not apply operator estimation to these observables

Breaking up a `Tensor` instance along these lines will now happen inside of PennyLane as the corresponding part of the `QNode` was also modified. This essentially means, that by the time the program execution hits this part, there will be no need for such checks here anymore.